src/content-items/handler.ts

```diff
@@ -75,12 +75,50 @@ export async function addEntryToDirectory(
     connection,
     parentPath,
     getRawDirectoryMetadataBytes(parentData),
-    wallet.privateKey,
+    wallet,
     podPassword,
-    metadataWithEpoch.epoch.getNextEpoch(getUnixTimestamp()),
+    getNextEpoch(metadataWithEpoch.epoch),
   )
 }
 
+/**
+ * Uploads magic word data on the next epoch's level
+ *
+ * Magic word should be uploaded if data is not found (in case of data pruning) or not deleted
+ */
+export async function deleteItem(
+  connection: Connection,
+  itemMetaPath: string,
+  wallet: utils.HDNode,
+  podPassword: PodPasswordBytes,
+): Promise<void> {
+  let pathInfo
+  try {
+    pathInfo = await getPathInfo(
+      connection.bee,
+      itemMetaPath,
+      prepareEthAddress(wallet.address),
+      connection.options?.requestOptions,
+    )
+    // eslint-disable-next-line no-empty
+  } catch (e) {}
+
+  // if the item already deleted - do nothing
+  if (pathInfo?.isDeleted) {
```

Comment:
it is not documented in the FIP. why do you need this when you have the "magic word" for indicating an element is deleted or not?